src/job/stagingJobHandler.ts

```diff
@@ -75,10 +77,35 @@ export class StagingJobHandler extends JobHandler {
     }
   }
   async deploy(): Promise<CommandExecutorResponse> {
+    let resp;
     try {
-      const resp = await this.deployGeneric();
+      // TODO: MONOREPO feature flag needed
+      if (
+        // process.env.FEATURE_FLAG_MONOREPO_PATH === 'true' &&
+        this.currJob.payload.repoName === MONOREPO_NAME
+      ) {
+        const logger = (message: string) => this.logger.save(this.currJob._id, message);
+        const repo_info = await this._docsetsRepo.getRepo(
+          this.currJob.payload.repoName,
+          this.currJob.payload.directory
+        );
+        if (!repo_info) {
+          const errorMessage = `
+            Docset Repo data not found in Atlas for repoName: ${this.currJob.payload.repoName}\n
+            ${this.currJob.payload.directory ? `directory: ${this.currJob.payload.directory}\n` : ''}
+            project: ${this.currJob.payload.project}
+          `;
+          logger(`ERROR: ${errorMessage}`);
+          throw Error(errorMessage);
+        }
+
+        const { bucket, url } = await this.getEnvironmentVariables();
+        resp = await nextGenStage({ job: this.currJob, logger, bucket, url });
+      } else {
+        resp = await this.deployGeneric();
+      }
       const summary = '';
-      if (resp?.output?.includes('Summary')) {
+      if (resp.output?.includes('Summary')) {
```

Comment:
the response object from `nextGenStage` doesn't have the `output` property. Instead, it has `outputText` and `errorText`.

Reply:
It does! I've modified `nextGenStage` to have the same response structure that `deployGeneric` does! We can always change this, though!

Reply:
Oh, sorry! I missed that. It's totally fine to keep it how it is. One thing I've noticed is that sometimes logs will be sent to `stderr` meaning that it will be in the `errorText` for some reason. I don't think it's an issue here, but if we don't see logs here at some point, that could be why.